log/DESIGN.md

```diff
@@ -0,0 +1,540 @@
+  as well as to resemble the behavior of getting tracers and meters.
+
+`Logger` can be extended by adding new `LoggerOption` options
+and adding new exported fields to the `LoggerConfig` struct.
+This design is already used in Trace API for getting tracers
+and in Metrics API for getting meters.
+
+Rejected alternative:
+
+- [Passing struct as parameter to LoggerProvider.Logger](#passing-struct-as-parameter-to-loggerproviderlogger).
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface:
+
+```go
+type Logger interface {
+	embedded.Logger
+	Emit(ctx context.Context, record Record)
+}
+```
+
+The specification may add new operations to `Logger`.
+The interface may have methods added without a package major version bump.
+This embeds `embedded.Logger` to help inform an API implementation
+author about this non-standard API evolution.
+This approach is already used in Trace API and Metrics API.
+
+### Logger.Emit
+
+The `Emit` method implements the [`Emit a LogRecord` operation](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord).
+
+[`Context` associated with the `LogRecord`](https://opentelemetry.io/docs/specs/otel/context/)
+is accepted as a `context.Context` method argument.
+
+Calls to `Emit` are supposed to be on the hot path.
+Therefore, in order to reduce the number of heap allocations,
+the [`LogRecord` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord),
+is defined as a `Record` type:
+
+```go
+type Record struct {
+	timestamp         time.Time
+	observedTimestamp time.Time
+	severity          Severity
+	severityText      string
+	body              string
+
+	// The fields below are for optimizing the implementation of
+	// attributes.
+	front [5]attribute.KeyValue
+	nFront int // The number of attributes in front.
+	back []attribute.KeyValue
+}
+```
+
+[`Timestamp`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-timestamp)
+is accessed using following methods:
+
+```go
+func (r Record) Timestamp() time.Time 
+func (r *Record) SetTimestamp(t time.Time) 
+```
+
+[`ObservedTimestamp`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-observedtimestamp)
+is accessed using following methods:
+
+```go
+func (r Record) ObservedTimestamp() time.Time 
+func (r *Record) SetObservedTimestamp(t time.Time) 
+```
+
+[`SeverityNumber`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber)
+is accessed using following methods:
+
+```go
+func (r Record) Severity() Severity
+func (r *Record) SetSeverity(s Severity)
+```
+
+`Severity` type is defined and constants are based on
+[Displaying Severity recommendation](https://opentelemetry.io/docs/specs/otel/logs/data-model/#displaying-severity).
+Additionally, `Severity[Level]1` constants are defined to make the API more readable and user friendly.
+
+```go
+type Severity int
```

Comment:
By @MadVikingGod:

> Because we will probably use the short name somewhere, I would suggest that we use stringer to make the short names.

I think it is a good idea to implement Stringer for all enum-like types. It would be helpful at least for debugging. I noticed that `slog` also implements stringer for "attribute value kinds".

Reply:
SIG meeting:
We can postpone the implementation of `String` even after making the Bridge API stable as it won't be a breaking change. Especially if we will have no agreement on the `string` representation.

Reply:
Addressed 6a47a3cdff9f2bdde41bcc1291d42d739a9b8a98